packages/runtime/src/next/client.ts

```diff
@@ -201,23 +230,51 @@ export class Makeswift {
     return response
   }
 
-  async getPages({
-    siteVersion = MakeswiftSiteVersion.Live,
-  }: {
-    siteVersion?: MakeswiftSiteVersion
-  } = {}): Promise<MakeswiftPage[]> {
-    const response = await this.fetch(`v3/pages`, siteVersion)
+  private getPagesInternal = async (
+    args: {
+      siteVersion?: MakeswiftSiteVersion
+      limit?: number
+      after?: string
+      sortBy?: 'title' | 'path' | 'createdAt' | 'updatedAt' | 'publishedAt'
+      sortDirection?: 'asc' | 'desc'
+      includeOffline?: boolean
+      pathPrefix?: string
+      locale?: string
+    } = {},
+  ) => {
+    const {
+      siteVersion = MakeswiftSiteVersion.Live,
+      limit = 20,
+      after,
+      includeOffline,
+      sortBy,
+      sortDirection,
+      pathPrefix,
+      locale,
+    } = args
+
+    const url = new URL('v4/pages', this.apiOrigin)
+    url.searchParams.set('limit', limit.toString())
+    if (after != null) url.searchParams.set('after', after)
+    if (includeOffline != null) url.searchParams.set('includeOffline', String(includeOffline))
+    if (sortBy != null) url.searchParams.set('sortBy', sortBy)
+    if (sortDirection != null) url.searchParams.set('sortDirection', sortDirection)
+    if (pathPrefix != null) url.searchParams.set('pathPrefix', pathPrefix)
+    if (locale != null) url.searchParams.set('locale', locale)
+
+    const response = await this.fetch(url.pathname + url.search, siteVersion)
```

Comment:
Why are we concatenating URL parts here? We can use `url.href` or `url.toString()`, or better yet, just pass the URL itself to `fetch`, it's supported. Or does `this.fetch` only support relative URLs as a string?

Reply:
`this.fetch` only accepts relative URLs, since it takes on the responsibility of constructing the URL using the `apiOrigin`